ibis/sql/alchemy.py

```diff
@@ -1009,7 +1106,8 @@ def list_tables(self, like=None, database=None, schema=None):
         tables : list of strings
         """
         inspector = self.inspector
-        names = inspector.get_table_names(schema=schema)
+        # inspector returns a mutable version of its names, so make a copy.
+        names = inspector.get_table_names(schema=schema).copy()
```

Comment:
A real bug in `list_tables`!

Reply:
Interesting! Those names are actually coming from `sqlalchemy`. Are the names it returns being reused inside of the inspector somehow?

Reply:
The issue I saw was with `names.extend` on the next line. Subsequent calls to `list_tables` were getting more and more copies of the views because `names` was a mutable list owned by the `inspector`. So by copying the list here, we get a fresh copy every time.

Reply:
Thanks for fixing this.